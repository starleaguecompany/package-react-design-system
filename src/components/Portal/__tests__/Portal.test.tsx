import * as React from 'react'
import { render, screen } from '@testing-library/react'

import { Portal } from '../../../index'

describe('Portal', () => {
  test('should render correctly with content', () => {
    render(<Portal>Test portal</Portal>)

    expect(screen.getByText('Test portal')).toMatchSnapshot()
    expect(screen.getByText('Test portal')).toBeVisible()
  })

  test('should be not a child of container element', () => {
    render(
      <div data-qa="container">
        <Portal>Test portal</Portal>
      </div>
    )

    expect(screen.getByText('Test portal')).toMatchSnapshot()
    expect(screen.getByTestId('container')).not.toContainElement(screen.getByText('Test portal'))
  })
})
