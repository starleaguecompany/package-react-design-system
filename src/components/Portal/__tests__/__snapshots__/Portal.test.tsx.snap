// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portal should be not a child of container element 1`] = `
<div
  sr-portal-container=""
>
  Test portal
</div>
`;

exports[`Portal should render correctly with content 1`] = `
<div
  sr-portal-container=""
>
  Test portal
</div>
`;
